### ISSUE TITLE
hydra-dispatch has a memory leaking issue

### ISSUE THREAD

**Issue by okupara** [nulab/hydralibs#14]
hydra-dispatch-react has a memory leaking issue because it internally updates state and developers have no way to cancel it.

The steps to reproduce is below.

1. fire dispatch for an async request when component mount or render
2. move to any paths before the request which sends at (1.) gets finished
3. react complains about memory leaking or an unnecessary rendering happens that it occurs memory leaking.

I created a test app to reproduce it.
https://codesandbox.io/s/eager-bhaskara-g6h08

There are two components they do dispatching the same dummy async request with Promise. "Test1" is using hydra-dispatch-react and "Test2" is using just only react-hooks.

When you push the "with hydra-dispatch" button, and then wait for seconds, you can see memory leak warning.

![ml](https://user-images.githubusercontent.com/68821/69511074-6f8e8e00-0f82-11ea-912a-98749e60ba98.gif)

On the other hand, after you reload the test app(because that warning shows up only one time), and push the "with hooks" button, and then wait for seconds, nothing happens, it means memory leak doesn't happen.

especially on developing SPA(applications have some url paths), this problem could annoy developers.

I think hydra-dispatch-react should have an option to cancel updating state.

**Comment by chaabaj** [comment]
Ok, I see, but the cancellation with react setState is done by checking the ref. I guess we could make aware dispatcherFromReact from the component reference and check it in the scheduler. It will do the same as Test2.tsx.

My guess is if you remove the code about checking the ref in Test2.tsx you would have the same issue.

**Comment by okupara** [comment]
I see.
> My guess is if you remove the code about checking the ref in Test2.tsx you would have the same issue.

Of course it does, my point is hydra-dispatch-react currently doesn't have any way to deal with that situation, but hooks does.

**Comment by chaabaj** [comment]
Yes so we need to allow dispatcherFromReact to track the reference
